google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStubSettings.java

```diff
@@ -0,0 +1,175 @@
+    super(builder);
+    instanceName = builder.instanceName;
+    appProfileId = builder.appProfileId;
+
+    // Per method settings.
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  /** Returns the target instance */
+  public InstanceName getInstanceName() {
+    return instanceName;
+  }
+
+  /** Returns the configured AppProfile to use */
+  public String getAppProfileId() {
+    return appProfileId;
+  }
+
+  /** Returns a builder containing all the values of this settings class. */
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  /** Builder for BigtableDataSettings. */
+  public static class Builder extends StubSettings.Builder<EnhancedBigtableStubSettings, Builder> {
+    private InstanceName instanceName;
+    private String appProfileId;
+
+    /**
+     * Initializes a new Builder with sane defaults for all settings.
+     *
+     * <p>Most defaults are extracted from BaseBigtableDataSettings, however some of the more
+     * complex defaults are configured explicitly here. Once the overlayed defaults are configured,
+     * the base settings are augmented to work with overlayed functionality (like disabling retries
+     * in the underlying GAPIC client for batching).
+     */
+    private Builder() {
+      this.appProfileId = SERVER_DEFAULT_APP_PROFILE_ID;
+
+      // Defaults provider
+      BigtableStubSettings.Builder baseDefaults = BigtableStubSettings.newBuilder();
+
+      setEndpoint(baseDefaults.getEndpoint());
+      setCredentialsProvider(baseDefaults.getCredentialsProvider());
+      setTransportChannelProvider(
+          InstantiatingGrpcChannelProvider.newBuilder()
+              // TODO: tune channels
+              .setChannelsPerCpu(2)
+              .setMaxInboundMessageSize(MAX_MESSAGE_SIZE)
+              .build());
+
+      // Per-method settings using baseSettings for defaults.
+    }
+
+    private Builder(EnhancedBigtableStubSettings settings) {
+      super(settings);
+      instanceName = settings.instanceName;
+      appProfileId = settings.appProfileId;
+
+      // Per method settings.
+    }
+
+    // <editor-fold desc="Private Helpers">
+    /**
+     * Copies settings from unary RPC to another. This is necessary when modifying request and
+     * response types while trying to retain retry settings.
+     */
+    private static void copyRetrySettings(
+        UnaryCallSettings.Builder<?, ?> source, UnaryCallSettings.Builder<?, ?> dest) {
+      dest.setRetryableCodes(source.getRetryableCodes());
+      dest.setRetrySettings(source.getRetrySettings());
+    }
+    // </editor-fold>
+
+    // <editor-fold desc="Public API">
+    /**
+     * Sets the target instance. This setting is required. All RPCs will be made in the context of
+     * this setting.
+     */
+    public Builder setInstanceName(InstanceName instanceName) {
+      Preconditions.checkNotNull(instanceName);
+      this.instanceName = instanceName;
```

Comment:
You can `this.instanceName = Preconditions.checkNotNull(instanceName)`. Up to you though.